src/extension/wps/wps-core/src/test/java/org/geoserver/wps/ppio/GeoJSONPPIOTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2020, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geoserver.wps.ppio;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.*;
+import java.util.List;
+import org.geoserver.config.GeoServer;
+import org.geoserver.config.GeoServerInfo;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.wps.WPSTestSupport;
+import org.geotools.data.DataUtilities;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.filter.text.cql2.CQL;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.io.WKTReader;
+import org.opengis.filter.Filter;
+
+public class GeoJSONPPIOTest extends WPSTestSupport {
+    private InputStream is;
+
+    @Override
+    protected void setUpSpring(List<String> springContextLocations) {
+        super.setUpSpring(springContextLocations);
+        springContextLocations.add("classpath*:/applicationContext-noargs.xml");
+    }
+
+    @Before
+    public void prepareInputStream() throws IOException {
+        is = SystemTestData.class.getResourceAsStream("states.json");
+        assertNotNull(is);
+    }
+
+    @After
+    public void cleanup() throws IOException {
+        if (is != null) {
+            is.close();
+        }
+    }
+
+    @Test
+    public void testEncodeGeometries() throws Exception {
+        WKTReader reader = new WKTReader();
+        Point point = (Point) reader.read("POINT(1.123456789 2.0)");
+
+        GeoServer gs = getGeoServer();
+        GeoServerInfo global = gs.getGlobal();
+        ByteArrayOutputStream os = new ByteArrayOutputStream(1024);
+        new GeoJSONPPIO.Geometries(gs).encode(point, os);
+        String output = os.toString();
+        assertEquals(output, "{\"type\":\"Point\",\"coordinates\":[1.12345679,2]}");
+
+        int dec = global.getSettings().getNumDecimals();
+        global.getSettings().setNumDecimals(4);
+        getGeoServer().save(global);
+
+        ByteArrayOutputStream os2 = new ByteArrayOutputStream(1024);
+        new GeoJSONPPIO.Geometries(gs).encode(point, os2);
+        String output2 = os2.toString();
+        assertEquals(output2, "{\"type\":\"Point\",\"coordinates\":[1.1235,2]}");
+
+        global.getSettings().setNumDecimals(dec);
```

Comment:
I would usually handle this in a try/finally block for extra safety...

Reply:
Even in a unit test?

Reply:
yeah, this was based on testing changes coordinate reference system configuration (so any failure produced so many other failures it was hard to track down.)

Keep in mind if you run into multi-threaded testing problems in the future.